Sming/SmingCore/Platform/System.cpp

```diff
@@ -31,16 +31,10 @@ void SystemClass::taskHandler(os_event_t* event)
 	}
 }
 
-void SystemClass::initialize()
+bool SystemClass::initialize()
 {
-	if(state != eSS_None)
-		return;
-	state = eSS_Intializing;
-
 	// Initialise the global task queue
-	system_os_task(taskHandler, USER_TASK_PRIO_1, taskQueue, TASK_QUEUE_LENGTH);
-
-	system_init_done_cb(staticReadyHandler);
```

Comment:
`system_init_done_cb` is used to guarantee that the callback will be executed once the system is ready( LWIP stack initialized + others). If you remove this line then where is that logic implemented?

Reply:
Task queue - we can create it in init() but it very much looks like SDK doesn't call queued functions until the ready - the debug dumps above appear to confirm that. I've tested this out and behaves normally. If you can think of any scenarios which this might break let me know and I'll test them?

Reply:
> but it very much looks like SDK doesn't call queued functions until it's ready

But this way we are relying on undocumented behaviour that might change with SDKs. IMHO it would be better to rely on  `system_init_done_cb`.

Reply:
> Or I could pop a question to Espressif to see what they say?

That will be a good start. Let's hope that they will respond. If starting the queuing mechanism is the last thing happening before `system_init_done_cb` is called then we are golden.

Reply:
Just to clarify, `StationClass::onSystemReady()` in the above post is a callback from the task queue, not `system_init_done_cb`.